src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesDeclarativeAgent.java

```diff
@@ -95,20 +107,124 @@ public ContainerTemplate getContainerTemplate() {
         return containerTemplate;
     }
 
+    @DataBoundSetter
+    public void setContainerTemplate(ContainerTemplate containerTemplate) {
+        this.containerTemplate = containerTemplate;
+    }
+
+    public PodTemplate getPodTemplate() {
+        return podTemplate;
+    }
+
+    @DataBoundSetter
+    public void setPodTemplate(PodTemplate podTemplate) {
+        this.podTemplate = podTemplate;
+        List<ContainerTemplate> containers = podTemplate.getContainers();
+        if (containers != null && containers.size() > 0) {
+            this.containerTemplate = containers.get(0);
+        }
+    }
+
     public int getActiveDeadlineSeconds() {
         return activeDeadlineSeconds;
     }
 
     @DataBoundSetter
     public void setActiveDeadlineSeconds(int activeDeadlineSeconds) { this.activeDeadlineSeconds = activeDeadlineSeconds; }
 
+    public String getPodTemplateName() {
+        return podTemplateName;
+    }
+
+    @DataBoundSetter
+    public void setPodTemplateName(String podTemplateName) {
+        this.podTemplateName = podTemplateName;
+        if (!StringUtils.isEmpty(podTemplateName)) {
+            String cloudName = getCloud();
+            if (StringUtils.isEmpty(cloudName)) {
+                cloudName = "kubernetes";
```

Comment:
Should use a constant, move `PodTemplateStep.DEFAULT_CLOUD` to `KubernetesCloud` and use that